docs/tutorial/building-a-theme.md

```diff
@@ -0,0 +1,1504 @@
+```
+
+Now, refactor the `layout.js` component in `gatsby-theme-events` to actually use Theme UI.
+
+First, import the `Layout`, `Header`, `Main`, and `Container` [components from Theme UI](https://theme-ui.com/layout).
+
+```javascript:title=gatsby-theme-events/src/components/layout.js
+import React from "react"
+// highlight-next-line
+import { Layout as ThemeLayout, Header, Main, Container } from "theme-ui"
+
+const Layout = ({ children }) => (
+  <div>
+    <h1>Gatsby Events Theme</h1>
+    {children}
+  </div>
+)
+
+export default Layout
+```
+
+Next, refactor the `layout.js` component to use the Theme UI components:
+
+```javascript:title=gatsby-theme-events/src/components/layout.js
+import React from "react"
+import { Layout as ThemeLayout, Header, Main, Container } from "theme-ui"
+
+// highlight-start
+const Layout = ({ children }) => {
+  return (
+    <ThemeLayout>
+      <Header>Gatsby Events Theme</Header>
+      <Main>
+        <Container>{children}</Container>
+      </Main>
+    </ThemeLayout>
+  )
+}
+// highlight-end
+
+export default Layout
+```
+
+Run the site to see the theme changes that are starting to take effect:
+
+```shell
+yarn workspace site develop
+```
+
+![Theme UI changes starting to take effect on the site. For example, the header is now purple.](./images/building-a-theme-theme-ui-changes.png)
+
+To continue applying theme styles, you can use the `Style` import from Theme UI. For example, in the `event-list.js` component, change the `<h1>`, `<ul>` and `<li>` elements to reference their themed styles:
+
+```javascript:title=gatsby-theme-events/src/components/event-list.js
+import React from "react"
+import { Link } from "gatsby"
+// highlight-next-line
+import { Styled } from "theme-ui"
+
+const EventList = ({ events }) => {
+  return (
+    <>
+      // highlight-next-line
+      <Styled.h1>Upcoming Events</Styled.h1>
+      // highlight-next-line
+      <Styled.ul>
+        {events.map(event => (
+          // highlight-next-line
+          <Styled.li key={event.id}>
+            <strong>
+              <Link to={event.slug}>{event.name}</Link>
+            </strong>
+            <br />
+            {new Date(event.startDate).toLocaleDateString("en-US", {
+              month: "long",
+              day: "numeric",
+              year: "numeric",
+            })}{" "}
+            in {event.location}
+            // highlight-next-line
+          </Styled.li>
+        ))}
+        // highlight-next-line
+      </Styled.ul>
+    </>
+  )
+}
+
+export default EventList
+```
+
+By replacing the `h1` with `Styled.h1`, `ul` with `Styled.ul`, and `li` with `Styled.li`, the theme styles for those elements have been applied:
+
+![Theme UI style changes showing on the events listing.](./images/building-a-theme-events-listing-styling.png)
+
+## Publish a theme to npm
+
+By publishing your theme to npm, you make it available to pull in as a dependency for any of your projects, and for anyone in the community to use, too.
```

Comment:
Is this required? Can we say something about private or local packages here in the event we get a question about it?

Reply:
I'm not sure what you mean by "required". It's required if you want to install your theme anywhere, or share it with the community. This tutorial is intended to mirror the Egghead course.

We can say it's not required _for the tutorial_ to publish the theme. I do reference that later, that they can use the published `@jlengstorf/gatsby-theme-events` if they're rather not publish it.

Reply:
Maybe something like this?

```suggestion
By publishing your theme to npm, you make it available to pull in as a dependency for any of your projects, and for anyone in the community to use, too.

> 💡 Please note that publishing your theme is only required if you want to install it on other sites or share it with the community. If you’re only building this theme as a learning exercise, you can skip this step.
```

Reply:
Perfecto